src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -85,18 +85,23 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
             // all methods that are compiled. Ideally the list of reflection enabled methods should be known before
             // we even start the compilation process (with the invocation stubs being compilation roots like any other).
             // The existing model has it's problems: e.g. the invocability of the method depends on inliner decisions.
-            if (factory.MetadataManager.HasReflectionInvokeStub(_method)
-                && !_method.IsCanonicalMethod(CanonicalFormKind.Any) /* Shared generics handled in the shadow concrete method node */)
+            if (factory.MetadataManager.HasReflectionInvokeStub(_method))
             {
                 if (dependencies == null)
                     dependencies = new DependencyList();
 
-                MethodDesc invokeStub = factory.MetadataManager.GetReflectionInvokeStub(Method);
-                MethodDesc canonInvokeStub = invokeStub.GetCanonMethodTarget(CanonicalFormKind.Specific);
-                if (invokeStub != canonInvokeStub)
-                    dependencies.Add(new DependencyListEntry(factory.FatFunctionPointer(invokeStub), "Reflection invoke"));
-                else
-                    dependencies.Add(new DependencyListEntry(factory.MethodEntrypoint(invokeStub), "Reflection invoke"));
+                if (!_method.IsCanonicalMethod(CanonicalFormKind.Any) /* Shared generics handled in the shadow concrete method node */)
+                {
+                    MethodDesc invokeStub = factory.MetadataManager.GetReflectionInvokeStub(Method);
+                    MethodDesc canonInvokeStub = invokeStub.GetCanonMethodTarget(CanonicalFormKind.Specific);
+                    if (invokeStub != canonInvokeStub)
+                        dependencies.Add(new DependencyListEntry(factory.FatFunctionPointer(invokeStub), "Reflection invoke"));
+                    else
+                        dependencies.Add(new DependencyListEntry(factory.MethodEntrypoint(invokeStub), "Reflection invoke"));
+                }
+
+                if (_method.OwningType.IsValueType && !_method.Signature.IsStatic)
+                    dependencies.Add(new DependencyListEntry(factory.MethodEntrypoint(_method, true), "Reflection unboxing stub"));
```

Comment:
I agree it looks odd here. I made the callsite explicitly name the parameter now.

As for making it an enum - most places actually use this like so:

```csharp
bool useUnboxingStub = method.OwningType.IsValueType && !method.Signature.IsStatic;
...
factory.MethodEntrypoint(method.GetCanonMethodTarget(CanonicalFormKind.Specific), useUnboxingStub);
```

I'm not sure making it an enum would lead to an improvement at those places. This is the only place that passes just a `true` without any sort of computation.